Sming/Libraries/ModbusMaster/component.mk

```diff
@@ -1,3 +1,4 @@
 COMPONENT_SUBMODULES := ModbusMaster
 COMPONENT_SRCDIRS := ModbusMaster/src
 COMPONENT_INCDIRS := ModbusMaster/src
+COMPONENT_CXXFLAGS := -DMB_RESPONSE_TIMEOUT_MS=300
```

Comment:
Try it like this:

```
# Configurable Modbus response timeout
COMPONENT_VARS += MB_RESPONSE_TIMEOUT
MB_RESPONSE_TIMEOUT ?= 300
COMPONENT_CXXFLAGS := -DMB_RESPONSE_TIMEOUT=$(MB_RESPONSE_TIMEOUT)
```

The value can then be changed via make-command line and will appear in `make list-config`.

Good practice to also document this value in `README.rst`:

```
.. envvar:: MB_RESPONSE_TIMEOUT

    description, etc.
```

The `envvar` directive allows this to be linkable from other .rst files:

```
You can change the timeout using :envvar:`MB_RESPONSE_TIMEOUT`
```

And will also appear in the built documentation index.

Reply:
Note: I'm using the value without `_MS` at the end as it's a little briefer and the documentation should provide details on how to use it anyway, but that's just a suggestion.

Reply:
Shorter is better, I began with the same, but thought that maybe making it obvious inside the variable would be better. I'll leave the shorter var.